src/MutableMap/MutableMap.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+package DafnyLibraries;
+
+import dafny.DafnySet;
+import dafny.DafnyMap;
+import dafny.Tuple2;
+import dafny.TypeDescriptor;
+
+import java.util.concurrent.*;
+import java.util.ArrayList;
+import java.util.Map;
+
+public class MutableMap<K,V> {
+  private ConcurrentHashMap<K,V> m;
+
+  public DafnyMap<K,V> Content() {
+    return new DafnyMap<K,V>(m);
+  }
+
+  public MutableMap() {
+    m = new ConcurrentHashMap<K,V>();
+  }
+
+  public MutableMap(TypeDescriptor<K> x, TypeDescriptor<V> y) {
+    m = new ConcurrentHashMap<K,V>();
+  }
+
+  public static <K,V> MutableMap<K,V> Make() {
+    return new MutableMap<K,V>();
+  }
+
+  public void Put(K k, V v) {
+    m.put(k, v);
+  }
+
+  public DafnySet<? extends K> Keys() {
+    return new DafnySet(m.keySet());
+  }
+
+  public boolean HasKey(K k) {
+    return m.containsKey(k);
+  }
+
+  public DafnySet<? extends V> Values() {
+    return new DafnySet(m.values());
+  }
+
+  public DafnySet<? extends Tuple2<K,V>> Items() {
+    ArrayList<Tuple2<K, V>> list = new ArrayList<Tuple2<K, V>>();
+    for (Map.Entry<K, V> entry : m.entrySet()) {
+      list.add(new Tuple2<K, V>(entry.getKey(), entry.getValue()));
+    }
+    return new DafnySet<Tuple2<K, V>>(list);
```

Comment:
As in `return new DafnySet(m.entrySet())`?

Reply:
I'll resolve this conversation as it's not applicable, but [`DafnyMap#entrySet`](https://github.com/dafny-lang/dafny/blob/e956ba4322e645f70f5e215c379ff78d50ae11b0/Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/DafnyMap.java#L153) has a nearly identical implementation so I thought there was a way to avoid the duplicate code with `return DafnyMap(m).entrySet()`. Turns out that wasn't possible because the `DafnyMap(HashMap)` constructor is private; and it isn't possible now because `m` is a `ConcurrentHashMap`.